### ISSUE TITLE
diff support for spatial columns and indexes

### ISSUE THREAD

**Issue by trevorr** [skeema/skeema#97]
As advertised, spatial columns and indexes are not yet supported:

```
2019-08-30 12:26:36 [WARN]  Skipping table `location_tz`: unable to generate DDL due to use of unsupported features. Use --debug for more information.
2019-08-30 12:26:36 [DEBUG] --- Expected CREATE
2019-08-30 12:26:36 [DEBUG] +++ MySQL-actual SHOW CREATE
2019-08-30 12:26:36 [DEBUG] @@ -3 +3 @@
2019-08-30 12:26:36 [DEBUG] -  `pt` point NOT NULL,
2019-08-30 12:26:36 [DEBUG] +  `pt` point NOT NULL /*!80003 SRID 4326 */,
2019-08-30 12:26:36 [DEBUG] @@ -8 +8 @@
2019-08-30 12:26:36 [DEBUG] -  KEY `ix_location_tz_pt` (`pt`(32))
2019-08-30 12:26:36 [DEBUG] +  SPATIAL KEY `ix_location_tz_pt` (`pt`)
```

It seems like it might not be too difficult to add this. That main issues seem to be the SRID on the column, the `SPATIAL` keyword in front of the key, and whatever is going on with that `(32)` after the key column.

**Comment by evanelias** [comment]
Thank you for the feature request! Afraid I have no prior familiarity with the spatial support in MySQL or MariaDB myself, as it's not a widely used feature yet. (InnoDB only gained support for spatial types relatively recently, so over the years many applications with spatial requirements just went with Postgres / PostGIS instead...)

To support spatial types in Skeema, I'd imagine a few potential points of complexity include:

* Researching any special properties of the 8 different spatial column types, and whether there are cases where it's safe/non-lossy to convert columns between these types, or even to/from non-spatial types
* Researching how SRIDs are represented in `information_schema`
* Researching any possible implementation differences between MySQL vs MariaDB, as well as version-specific differences
* Building integration test cases that accurately cover all of the above

Realistically, given my own lack of experience with spatial data, support in Skeema will require either third-party code contribution, or a corporation willing to sponsor the development time.

**Comment by evanelias** [comment]
SPATIAL index diff support is being merged to main branch momentarily. This works for all database flavors/versions supporting InnoDB SPATIAL indexes: MySQL 5.7+ and MariaDB 10.2+. This functionality will be included in the next release (Skeema v1.11.0), planned for September.

Ditto for MySQL 8's SRID support, which was merged to main yesterday in commit 5e357db.

MariaDB's equivalent syntax (REF_SYSTEM_ID) is only semi-supported though, because MariaDB doesn't ever expose that attribute in SHOW CREATE TABLE, so it unfortunately gets lost upon dumping a table/schema (including by `skeema init` or `skeema pull`).

**Comment by evanelias** [comment]
Skeema v1.11.0 was [released today](https://www.skeema.io/blog/2023/09/26/v1110-released/), containing full support for SPATIAL indexes, as well as MySQL 8's SRID column attribute.

Users of AWS Aurora v2 (MySQL 5.7 compatible), please note that Aurora's [nonstandard spatial index implementation](https://aws.amazon.com/blogs/database/amazon-aurora-under-the-hood-indexing-geospatial-data-using-z-order-curves/) unfortunately diverges from standard MySQL's `information_schema` behavior. As a result, spatial indexes in Aurora v2 are unsupported in [Skeema community edition](https://www.skeema.io/docs/requirements/#aws-aurora). A special workaround for this problem is implemented in [Skeema premium edition](https://www.skeema.io/download/).

AWS Aurora v3 (MySQL 8.0 compatible) appears to fix this problem, matching standard MySQL `information_schema` behavior for spatial indexes. Spatial indexes in Aurora v3 work correctly in both editions of Skeema.
